### PR TITLE
Bringup: Nothing Phone 1 (Spacewar)

### PR DESCRIPTION
**Pull request by fekz115**
Here is dmesg: 
https://gist.github.com/fekz115/35123230f2a2640b1680c5208343c910

Still didn't managed to enable pmic-glink. Help wanted.
Tried to set PM7250B_SID and PM7250B_SID1 to every values from 0 to 9.
Here is my stock fdt: https://gist.github.com/fekz115/3ff79713e2d0d6caa91ed12eaa7657d9
Here is dmesg with PM7250B_SID and PM7250_SID1 set 2 and 3:
https://gist.github.com/fekz115/d6b1f2ab691e6b14ada1cfd5803266c1
Seems like here is something bad happened: https://gist.github.com/fekz115/d6b1f2ab691e6b14ada1cfd5803266c1#file-gistfile1-txt-L299
Set compatible (qcom,sm7325-pmic-glink or whatever) also doesn't work

State:

- [x] UFS
- [x] USB
- [x] Simple Framebuffer 
- [x] GPU + Panel
- [x] Touchscreen (focaltech fts (spi))
- [x] Vol+, Vol-, Power buttons
- [x] Most of remoteprocs
- [ ] Wifi (partially)
- [ ] IPA (partially)
- [x] Flash led 
- [ ] I2C usb switcher
- [ ] Sound (need to port tfa98xx driver from downstream)
- [ ] Thermal Zones
- [ ] Bluetooth
- [ ] Battery/Charging
- [ ] Cameras

Notes:
- My home wifi has 5Gz and 2.4Gz access points, and it connects only to 2.4Gz and fails on timeout for 5Gz, but it can connect to other 5Gz networks. (downstream can connect to my home wifi too).
- IPA modem shown in specific soft as mm or modem-manager-gui, but in not ready state. Also, phosh reports that there is no modem, plasma-mobile show me SIM info, even operator APN, but can't connect to network.

### CODE REVIEW

**Comment by fekz115** [comment]
@z3ntu In case you did not receive a notification of my message to the matrix, PR is ready for review. Let me know when you have a chance to take a look.

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+
+		key-volume-up {
+			label = "Volume up";
+			gpios = <&pm7325_gpios 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEUP>;
+		};
+	};
+
+	pmic-glink {
+		compatible = "qcom,pmic-glink";
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:47]: Needs second compatible, see docs. But not sure if it's a better idea to add qcom,sm7325-pmic-glink or re-use qcom,qcm6490-pmic-glink. Probably the first one?

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+#include "pm8350c.dtsi" /* PM7350C */
+#include "pmk8350.dtsi" /* PMK7325 */
+
+/delete-node/ &rmtfs_mem;
+
+/ {
+	model = "Nothing Phone (1)";
+	compatible = "nothing,spacewar", "qcom,sm7325";
+	qcom,msm-id = <0x1db 0x10000 0x203 0x10000>;
+	qcom,board-id = <0x1000b 0x00>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:30]: You can probably remove these two properties when you use bootimg header version 2

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 3]
```diff
@@ -0,0 +1,642 @@
...
+			reg = <0x0 0xc0000000 0x0 0x5100000>;
+			no-map;
+		};
+
+		framebuffer_region@e1000000 {
+			reg = <0x0 0xe1000000 0x0 (1080 * 2400 * 4)>;
+			no-map;
+		};
+
+		rmtfs_mem: memory@f8500000 {
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:109]: rmtfs@

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 4]
```diff
@@ -0,0 +1,642 @@
...
+			reg = <0x0 0x88f00000 0x0 0x1e00000>;
+			no-map;
+		};
+
+		removed_mem: removed@c0000000 {
+			reg = <0x0 0xc0000000 0x0 0x5100000>;
+			no-map;
+		};
+
+		framebuffer_region@e1000000 {
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:104]: no underscore in node names, you can just make this framebuffer@ or cont-splash@ like in other devices

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 5]
```diff
@@ -0,0 +1,642 @@
...
+&gpi_dma1 {
+	status = "okay";
+};
+
+&gpu {
+	status = "okay";
+};
+
+&gpu_zap_shader {
+	firmware-name = "qcom/sm7325/nothing-spacewar/a660_zap.mbn";
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:367]: You should check what the latest firmware directory naming way is, I think `qcom/sm7325/nothing/spacewar/<foo>` is the one used now..

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 6]
```diff
@@ -0,0 +1,642 @@
...
+	};
+
+	ts_reset_suspend: ts_reset_suspend {
+		pins = "gpio105";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-down;
+	};
+
+	ts_active: ts_active {
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:573]: Even though they're not going upstream now, please align the node naming with the bindings

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 7]
```diff
@@ -0,0 +1,642 @@
...
+		pins = "gpio105", "gpio81";
+		function = "gpio";
+		drive-strength = <8>;
+		bias-pull-up;
+	};
+
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:579]: remove extra newline

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 8]
```diff
@@ -0,0 +1,642 @@
...
+	status = "okay";
+};
+
+&usb_1 {
+	status = "okay";
+};
+
+&usb_1_dwc3 {
+	dr_mode = "peripheral";
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:611]: you already have pmic-glink in, is otg/role switch not working?

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 9]
```diff
@@ -0,0 +1,642 @@
...
+&usb_1_dwc3_hs {
+	remote-endpoint = <&pmic_glink_hs_in>;
+};
+
+&usb_1_dwc3_ss {
+	remote-endpoint = <&pmic_glink_ss_in>;
+};
+
+&usb_1_hsphy {
+	status = "okay";
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:623]: status should be last property

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 10]
```diff
@@ -0,0 +1,642 @@
...
+	vdda-pll-supply = <&vreg_l10c>;
+	vdda18-supply = <&vreg_l1c>;
+	vdda33-supply = <&vreg_l2b>;
+};
+
+&usb_1_qmpphy {
+	status = "okay";
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:630]: status should be last property

File: arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts [hunk 11]
```diff
@@ -0,0 +1,642 @@
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/sm7325-nothing-spacewar.dts:1]: The commit message format doesn't match the subsystem, should be `arm64: dts: qcom: foobar`

Also you need to add some extra lines as commit message, not just subject. I'd write something like this:

```
arm64: dts: qcom: Add Nothing Phone (1)

Add an initial dts for the Nothing Phone (1) based on the SM7325 SoC.
This SoC is, just like QCM6490, mostly compatible with SC7280 dtsi so create a SM7325 SoC based on that.

Working features:
* list the
* working features

And maybe some more text that you want to add, anything interesting you can think of
```

File: drivers/gpu/drm/msm/dsi/dsi_host.c
Comment (z3ntu) [inline drivers/gpu/drm/msm/dsi/dsi_host.c:1]: wrong commit message subject, see `git log drivers/gpu/drm/msm/dsi/dsi_host.c`

**Comment by z3ntu** [comment]
Along with the commits I added, please check the commit subjects for your commits, they don't match the subsystem in many cases. Just check `git log --oneline path/to/dir`

Also please run `make CHECK_DTBS=1 qcom/sm7325-your-phone.dtb` and make sure this doesn't throw any errors (there are some coming from dtsi with pci@ or whatever, you can ignore those)

Regarding pm7250b, there doesn't seem to be a trace of it in downstream dts, why do you think your phone has a pm7250b?

**Comment by fekz115** [comment]
Will be updated by another person.

**Comment by z3ntu** [comment]
I'd love to see this getting finished up!

**Pull request closed without merging**